core/ledger/ledgermgmt/ledger_mgmt.go

```diff
@@ -132,6 +167,26 @@ func (m *LedgerMgr) CreateLedgerFromSnapshot(snapshotDir string) (ledger.PeerLed
 	}, cid, nil
 }
 
+func (m *LedgerMgr) setUnderCreationLedger(snapshotDir string) error {
+	m.underCreationLock.Lock()
+	defer m.underCreationLock.Unlock()
+	if m.underCreationLedger != "" {
+		return errors.New("a ledger is being created from a snapshot. Call 'peer channel joinbysnapshotstatus' to check the status and execute the command again after it is done.")
```

Comment:
I don't think that putting actual command `peer channel joinbysnapshotstatus` here is a good idea. If something changes in the command, no one will know that has to be fixed here.

Reply:
agreed